dbt/dbt_project.yml

```diff
@@ -36,6 +36,9 @@ vars:
   # frames for testing
   test_qc_year_end: 2030
 
+  # Township code to use for filtering town close QC reports
+  qc_report_town_code: "71"  # Lake Township
```

Comment:
I figure we can use the `qc_report_*` prefix to mark symbols that are used for reports, and if we wind up with multiple reports that need different town codes we can modify this variable to make it more specific.

Note that setting a variable isn't the most ergonomic way of accomplishing this goal (filtering reports by town) but I think it's the fastest path forward given dbt's design. The upside is that dbt exposes a built-in interface for variable overrides that we can use during report generation (via the `--vars` flag); the downside is that since the value is baked into the view, we have to rebuild any views that rely on this variable if we want to override it during export. See the script below for some details about how we handle this using the `--vars` and `--rebuild` flags. This is a little bit annoying, but I think it should be fine assuming A) none of our QC views take an obscene amount of time to execute and B) we continue following the historical pattern of generating QC reports once per town on town close.

An alternative might be to keep the underlying view unfiltered by township, and implement a `--township` option on the script that performs the filtering via query (similar to the way the `transform_dbt_test_results.py` script works). This would simplify the underlying view and make it more flexible, but in so doing would move more complexity into the export script and would tightly couple the export script to the notion of township filters, which may not apply to all exports that we end up running.  I feel somewhat agnostic on this direction but happy to move towards it if you feel strongly! I'll note that this alternative is probably more closely aligned with a potential future where reports are self-service and run frequently on demand, since in that future we might open ourselves up to race conditions if we're rebuilding models every time a use requests a report filtered by a township.

Reply:
I added a `--where` filter in bce0bfe27f8c2d57d674d7a6b1744341639835bf and I think I like it better! Let me know what you think, happy to modify the design if you have a better idea.